File: examples/transforms/status_up_reject_pending.js
#!/usr/bin/env node
"use strict";

const input = readJson();

if (input.status === "active") {
  writeJson({ state: "enabled" });
} else if (input.status === "suspended") {
  writeJson({ state: "disabled" });
} else if (input.status === "closed") {
  writeJson({ state: "archived" });
} else if (input.status === "pending_review") {
  process.stderr.write("unsupported status: pending_review\n");
  process.exitCode = 12;
} else {
  process.stderr.write(`unknown status: ${input.status}\n`);
  process.exitCode = 12;
}

function readJson() {
  return JSON.parse(require("node:fs").readFileSync(0, "utf8"));
}

function writeJson(value) {
  process.stdout.write(`${JSON.stringify(value)}\n`);
}

File: examples/transforms/status_up_reject_pending.py
#!/usr/bin/env python3
import json, sys
x = json.load(sys.stdin)
status = x.get("status")
if status == "active":
    print(json.dumps({"state": "enabled"}))
elif status == "suspended":
    print(json.dumps({"state": "disabled"}))
elif status == "closed":
    print(json.dumps({"state": "archived"}))
elif status == "pending_review":
    print("unsupported status: pending_review", file=sys.stderr)
    sys.exit(12)
else:
    print(f"unknown status: {status}", file=sys.stderr)
    sys.exit(12)
